fix: don't mutate the bounded text if not updated when submitted

ad1992 opened the pull request:
Attempt fix the "undo twice for first time when bounded text properties updated by selecting text explicitly" mentioned in https://github.com/excalidraw/excalidraw/pull/4537

Previous
![Excalidraw (12)](https://user-images.githubusercontent.com/11256141/148205106-dd099698-39e9-49e4-861c-0239be882d69.gif)

Now


![Excalidraw (13)](https://user-images.githubusercontent.com/11256141/148205406-a1f6f2ba-55a7-4900-a0f8-4ba063b02c03.gif)

dwelle commented on src/element/newElement.ts:
```diff
@@ -246,11 +246,17 @@ export const updateTextElement = (
+      ? undefined
+      : getAdjustedDimensions(element, text);
+
+  text = boundToContainer && !isSubmit ? element.text : text;
```
why updating `text` to stale state while editing bound text element?

ad1992 replied:
I don't want to update text until submitted so that before submit its possible to compare old text and new text and check if mutation is needed as done in line 457 in `textwysiswg.tsx`

dwelle replied:
This creates complexity and exposing details all over the codebase that shouldn't be exposed. Also, it breaks collaboration (latest value is not synced to remote clients while editing), and may break other things (now or in the future). Lastly, it's not clear from the code what this does at all, so would require extensive comments and such.

In any case, we can just keep the original text as a local state within the wysiwyg editor closure, no?

dwelle commented on package.json:
```diff
@@ -77,7 +77,7 @@
   },
   "engines": {
-    "node": ">=14.0.0"
+    "r": ">=14.0.0"
```
r?

ad1992 replied:
typo :/ , thanks for catching this

ad1992 commented on package.json:
```diff
@@ -77,7 +77,7 @@
   },
   "engines": {
-    "node": ">=14.0.0"
+    "r": ">=14.0.0"
```
```suggestion
    "node": ">=14.0.0"
```

dwelle commented on src/element/textWysiwyg.tsx:
```diff
@@ -566,6 +567,12 @@ export const textWysiwyg = ({
+      const updatedElement = Scene.getScene(element)!.getElement(
+        element.id,
+      ) as ExcalidrawTextElement;
+      prevText = updatedElement.text;
```
doesn't this assignment make the `prevText` variable equivalent to `updatedElement.text`?

dwelle commented:
I noticed a regression:

1. Create a sticky note and write line long enough to wrap one the next line. Confirm.
2. Edit again and delete the text, then write a single-line text. Confirm.
3. The text is misaligned.

	![excal-regr-sticky](https://user-images.githubusercontent.com/5153846/148832258-f76f7a9f-656f-42a7-aa8e-ff3029848b85.gif)

ad1992 commented:
> I noticed a regression:
> 
> 1. Create a sticky note and write line long enough to wrap one the next line. Confirm.
> 2. Edit again and delete the text, then write a single-line text. Confirm.
> 3. The text is misaligned.
>    ![excal-regr-sticky](https://user-images.githubusercontent.com/5153846/148832258-f76f7a9f-656f-42a7-aa8e-ff3029848b85.gif)

Good catch!. Updated the implementation to cache `originalText` and comparing with editor value instead. Added spec for this case. Should be fine now.

dwelle commented:
🚀
